### PR TITLE
Word Generator

### PR DESCRIPTION
**Pull request by arya-hat**
# Summary
Implemented word generation logic.

# Details
The structure of the implementation is as follows:
public/
├── single_words.txt
└── multi_words.txt

src/
└── wordGeneration/
    ├── generators/
    │   ├── SingleWordGenerator.ts
    │   ├── MultiWordGenerator.ts
    │   ├── SingleWordGenerator.test.ts
    │   └── MultiWordGenerator.test.ts
    │
    └── index.ts

**SingleWordGenerator.ts:** : This file contains the `SingleWordGenerator` class that contains two functions. The first function: `loadWords()` fetches the .txt file which contains the word bank we are outputting from. The file is cleaned so the words can be stored in an array. The second function: `getTypingPrompt()` just calls loadWords(), calculates a random index and returns the word with that index in the array.

**MultiWordGenerator.ts**: This file contains the `MultWordGenerator` class that also contains two functions. This class takes in a `minWords` and `maxWords` parameter which is are instance variables. The first function: `loadWords()` acts the exact same way as the first file. The second function: `getTypingPrompt()` also calls the first function. Utilizing `minWords` and `maxWords` it determines a random word count that falls within the specified range. While the result is shorter than the desired count we compute the random index the same way we did in the first file, output the word, and ensure the word has not been already outputted in the same instance. We add this word to our array and join it in the end to output multiple words separated by spaces.

# Testing
In both test files we mock the fetch functionality using `jest.spyOn`. 

There are two separate test suites, one for each class. They both contain the necessary unit tests to ensure functionality, error handling, and edge cases. 
image

### CODE REVIEW

File: src/wordGeneration/wordGenerator.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import {
+  ENGLISH_WORD_LENGTH,
+  ENGLISH_WORD_COUNT,
+} from "./config";
```
Comment (newracket) [inline src/wordGeneration/wordGenerator.ts]: Word length/word count shouldn't be harcoded, this should be a parameter for the function. If another component wants to call your function but wants to allow words with a length of 6, they can't do that.

File: src/wordGeneration/wordGenerator.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+/**
+ * Returns multiple random English words by running getRandomEnglishWord() in a loop.
+ * Enforces word count constraints.
+ */
+export function getRandomEnglishWords(
+  count: number,
+  options: WordGeneratorOptions = {}
+): string[] {
+  if (
```
Comment (newracket) [inline src/wordGeneration/wordGenerator.ts]: What's the point of this, why have this limitation? If the caller of this function wants to generate 20 words, why limit them?

File: src/wordGeneration/wordGenerator.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    count > ENGLISH_WORD_COUNT.MAX
+  ) {
+    throw new Error(
+      `Word count must be between ${ENGLISH_WORD_COUNT.MIN} and ${ENGLISH_WORD_COUNT.MAX}.`
+    );
+  }
+
+  const words: string[] = [];
+
+  for (let i = 0; i < count; i++) {
```
Comment (newracket) [inline src/wordGeneration/wordGenerator.ts]: Not sure if this is intended, but this logic would allow for there to be duplicates within the words array.

File: src/wordGeneration/wordGenerator.test.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      rng: () => 0,
+    });
+
+    expect(words).toEqual([
+      customWords[0],
+      customWords[0],
+    ]);
+  });
+
+  it("throws an error when no valid words exist", () => {
```
Comment (newracket) [inline src/wordGeneration/wordGenerator.test.ts]: This behavior doesn't make sense to me. If the caller of the function provides a list, there's no reason why the function arbitrarily decides to filter the list.
